src/autocomplete/databases/postgresql/postgresql-extract-statements.ts

```diff
@@ -0,0 +1,102 @@
+import {Token, tokenize} from '../../shared/tokenize';
+import {postgreSqlAutocompleteData} from './postgresql-autocomplete';
+import {PostgreSqlLexer} from './generated/PostgreSqlLexer';
+import {CursorPosition} from '../../shared';
+
+export interface PostgreSqlStatement {
+    startLine: number;
+    endLine: number;
+    startColumn: number;
+    endColumn: number;
+    statement: string;
+    hasCursor?: boolean;
+}
+
+export function extractPostgreSqlStatementsFromQuery(
+    query: string,
+    cursorPosition?: CursorPosition,
+): PostgreSqlStatement[] {
+    const {tokens} = tokenize(
+        postgreSqlAutocompleteData.Lexer,
+        PostgreSqlLexer.symbolicNames,
+        postgreSqlAutocompleteData.tokenDictionary.SPACE,
+        query,
+    );
+
+    let startLine = 0;
+    let startColumn = 0;
+    let statementStartIndex = 0;
+    let pendingNewStatement = true;
+    let lastStatementToken: Token;
```

Comment:
I don't like that `statementStartIndex` and `statementLastIndex` have different types. Let's make them all indexes and name them accordingly `(last|first)StatementTokenIndex`

Reply:
I don't know why you compare this variables if they have different duties, just look how I use `lastStatementToken`